spec/abstractops.html

```diff
@@ -1440,24 +1440,23 @@ <h1>
       <dt>description</dt>
       <dd>It parses the argument as an ISO 8601 duration string and returns a Duration Record.</dd>
     </dl>
-    <emu-note>The value of ToIntegerWithTruncation(*""*) is 0.</emu-note>
     <emu-note>Use of mathematical values rather than approximations is important to avoid off-by-one errors with input like "PT46H66M71.50040904S".</emu-note>
     <emu-alg>
       1. Let _duration_ be ParseText(StringToCodePoints(_isoString_), |TemporalDurationString|).
       1. If _duration_ is a List of errors, throw a *RangeError* exception.
       1. Let each of _sign_, _years_, _months_, _weeks_, _days_, _hours_, _fHours_, _minutes_, _fMinutes_, _seconds_, and _fSeconds_ be the source text matched by the respective |Sign|, |DurationYears|, |DurationMonths|, |DurationWeeks|, |DurationDays|, |DurationWholeHours|, |DurationHoursFraction|, |DurationWholeMinutes|, |DurationMinutesFraction|, |DurationWholeSeconds|, and |DurationSecondsFraction| Parse Node contained within _duration_, or an empty sequence of code points if not present.
-      1. Let _yearsMV_ be ? ToIntegerWithTruncation(CodePointsToString(_years_)).
-      1. Let _monthsMV_ be ? ToIntegerWithTruncation(CodePointsToString(_months_)).
-      1. Let _weeksMV_ be ? ToIntegerWithTruncation(CodePointsToString(_weeks_)).
-      1. Let _daysMV_ be ? ToIntegerWithTruncation(CodePointsToString(_days_)).
-      1. Let _hoursMV_ be ? ToIntegerWithTruncation(CodePointsToString(_hours_)).
+      1. If _years_ is *undefined*, let _yearsMV_ be 0; else let _yearsMV_ be ? ToIntegerWithTruncation(CodePointsToString(_years_)).
+      1. If _months_ is *undefined*, let _monthsMV_ be 0; else let _monthsMV_ be ? ToIntegerWithTruncation(CodePointsToString(_months_)).
+      1. If _weeks_ is *undefined*, let _weeksMV_ be 0; else let _weeksMV_ be ? ToIntegerWithTruncation(CodePointsToString(_weeks_)).
+      1. If _days_ is *undefined*, let _daysMV_ be 0; else let _daysMV_ be ? ToIntegerWithTruncation(CodePointsToString(_days_)).
+      1. If _hours_ is *undefined*, let _hoursMV_ be 0; else let _hoursMV_ be ? ToIntegerWithTruncation(CodePointsToString(_hours_)).
       1. If _fHours_ is not empty, then
         1. If any of _minutes_, _fMinutes_, _seconds_, _fSeconds_ is not empty, throw a *RangeError* exception.
         1. Let _fHoursDigits_ be the substring of CodePointsToString(_fHours_) from 1.
         1. Let _fHoursScale_ be the length of _fHoursDigits_.
         1. Let _minutesMV_ be ? ToIntegerWithTruncation(_fHoursDigits_) / 10<sup>_fHoursScale_</sup> &times; 60.
       1. Else,
-        1. Let _minutesMV_ be ? ToIntegerWithTruncation(CodePointsToString(_minutes_)).
+        1. If _minutes_ is *undefined*, let _minutesMV_ be 0; else let _minutesMV_ be ? ToIntegerWithTruncation(CodePointsToString(_minutes_)).
```

Comment:
I think we can keep ParseTemporalDurationString as it was. In the case where one of the parse nodes is not present (for example, let's say _years_):
- _years_ becomes an empty sequence of code points
- CodePointsToString(_years_) is the empty string **""**
- [ToIntegerWithTruncation](https://tc39.es/proposal-temporal/#sec-tointegerwithtruncation)(**""**) calls [ToNumber](https://tc39.es/ecma262/#sec-tonumber)(**""**)
- ToNumber step 6, calls [StringToNumber](https://tc39.es/ecma262/#sec-stringtonumber)(**""**)
- StringToNumber parses an empty sequence of code points against [_StringNumericLiteral_](https://tc39.es/ecma262/#prod-StringNumericLiteral), which matches _StrWhiteSpace_(opt)
- [StringNumericValue](https://tc39.es/ecma262/#sec-runtime-semantics-stringnumericvalue) of _StrWhiteSpace_(opt) is **+0**<sub>𝔽</sub>
- ToIntegerWithTruncation returns truncate(ℝ(**+0**<sub>𝔽</sub>)) which is 0.

(Unlike the polyfill code where we do need to check for undefined because we're using a regexp)